e2e/backend_service_update_test.go

```diff
@@ -0,0 +1,225 @@
+	}
+}
+func TestBackendServiceUpdate(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	graphBuilder := rgraph.NewBuilder()
+
+	meshURL, meshKey := ensureMesh(ctx, t)
+	t.Cleanup(func() {
+		err := theCloud.Meshes().Delete(ctx, meshKey)
+		t.Logf("theCloud.Meshes().Delete(ctx, %s): %v", meshKey, err)
+	})
+
+	hc1ID, err := buildHealthCheck(graphBuilder, "hc1-test", 15)
+	if err != nil {
+		t.Fatalf("buildHealthCheck(_, hc1-test, 15) = (_, %v), want (_, nil)", err)
+	}
+	hc2ID, err := buildHealthCheck(graphBuilder, "hc2-test", 15)
+	if err != nil {
+		t.Fatalf("buildHealthCheck(_, hc2-test, 15) = (_, %v), want (_, nil)", err)
+	}
+
+	bs1Name := resourceName("bs1-e2e")
+	bs2Name := resourceName("bs2-e2e")
+
+	bs1ID, err := buildBackendServiceWithLBScheme(graphBuilder, bs1Name, hc1ID, "INTERNAL_SELF_MANAGED")
+	if err != nil {
+		t.Fatalf("buildBackendServiceWithLBScheme(_, %s, _) = %v, want nil", bs1Name, err)
+	}
+
+	bs2ID, err := buildBackendServiceWithLBScheme(graphBuilder, bs2Name, hc2ID, "INTERNAL_SELF_MANAGED")
+	if err != nil {
+		t.Fatalf("buildBackendServiceWithLBScheme(_, %s, _) = %v, want nil", bs2Name, err)
+	}
+
+	rules := []*networkservices.TcpRouteRouteRule{
+		{
+			Action: &networkservices.TcpRouteRouteAction{
+				Destinations: []*networkservices.TcpRouteRouteDestination{
+					{
+						ServiceName: resourceSelfLink(bs1ID),
+						Weight:      10,
+					},
+				},
+			},
+			Matches: []*networkservices.TcpRouteRouteMatch{
+				{
+					Address: routeCIDR,
+					Port:    "80",
+				},
+			},
+		},
+		{
+			Action: &networkservices.TcpRouteRouteAction{
+				Destinations: []*networkservices.TcpRouteRouteDestination{
+					{
+						ServiceName: resourceSelfLink(bs2ID),
+						Weight:      10,
+					},
+				},
+			},
+			Matches: []*networkservices.TcpRouteRouteMatch{
+				{
+					Address: routeCIRD2,
+					Port:    "80",
+				},
+			},
+		},
+	}
+	tcpr, err := buildTCPRoute(graphBuilder, "test-route", meshURL, rules, bs1ID)
+	if err != nil {
+		t.Fatalf("buildTcpRoute(_, test-route, %s, %v, %s) = %v, want nil", meshURL, rules, bs1ID, err)
+	}
+
+	t.Logf("tcpr = %s", pretty.Sprint(tcpr))
+
+	graph, err := graphBuilder.Build()
+	if err != nil {
+		t.Fatalf("graphBuilder.Build() = %v, want nil", err)
+	}
+
+	result, err := plan.Do(ctx, theCloud, graph)
+	if err != nil {
+		t.Fatalf("plan.Do(_, _, _) = %v, want nil", err)
+	}
+
+	ex, err := exec.NewSerialExecutor(result.Actions)
+	if err != nil {
+		t.Fatalf("exec.NewSerialExecutor(_, _) err: %v", err)
+		return
+	}
+	res, err := ex.Run(ctx, theCloud)
+	if err != nil || res == nil {
+		t.Errorf("ex.Run(_,_) = %v, want nil", err)
+	}
+	t.Cleanup(func() {
+		err := theCloud.TcpRoutes().Delete(ctx, tcpr.Key)
+		if err != nil {
+			t.Logf("delete TCProute: %v", err)
+		}
+		err = theCloud.BackendServices().Delete(ctx, bs1ID.Key)
+		if err != nil {
+			t.Logf("delete backend service: %v", err)
+		}
+		err = theCloud.BackendServices().Delete(ctx, bs2ID.Key)
```

Comment:
you need to clean up health checks also (after backend services)

Reply:
Done.